README.md

```diff
@@ -1,2 +1,73 @@
-# declarative-custom-elements
-Explainer draft for specifying an approach to Declarative Custom Elements
+# Declarative Custom Elements
+### A discussion...
+
+## Problem
+Imperative JS code is required to unlock the power of [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements), [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), and other "web components" APIs. This is particularly true when these APIs are used in concert with each other. A declarative API would open the door for things like:
+
+- non-JS developers being given the power of custom elements
+- having custom element definitions available as parse time without requireing the jump over to the JS thread early in the document parse
+- others?
+
+*Questions:* Have we come to a future where "web components" amount to _just_ Custom Elements and Shadow DOM? Could we benefit from including other specs in this conversation? Possibly:
+
+- [constructible stylesheets](https://web.dev/constructable-stylesheets/)
+- [adoptedStyleSheets](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets)
+- [aria delegation](https://leobalter.github.io/cross-root-aria-delegation/) (possibly implied but also not currently "spec", only proposal)
+- [scoped registries](https://github.com/WICG/webcomponents/issues/716)
+- others? see [current Web Components Community Group priorities](https://w3c.github.io/webcomponents-cg/)
+
+## Existing research
+*NOTE:* It would be great to get someone with specific experience with these technologies to write up a demo outlining the declarative nature of components in their context.
+
+- Vue
+- Svelte
+- [Declarative Shadow DOM](https://web.dev/declarative-shadow-dom/)
+- [Declarative CSS modules](https://github.com/WICG/webcomponents/issues/939)
+- [x-element](https://www.npmjs.com/package/astro-xelement) from Astro
+- [decorate element](https://matthewphillips.info/programming/posts/decorate-element/)
+  - cons: much like DSD does not allow for scripting.
+- others?
+
+Previous conversation in this space have halted at "do we have enough coverage of exisiting paradigms in the room?", let's address that early and often!
+
+## Dreams
+I would could this as a smashing success if we were able to do something _like_ the following, but understand this is more of a starting place than necessarily the API that we should drive towards.
+
+1. *External "module"*
+  ```html
+    <import rel="customelement" href="my-element.html" tagname="my-element" />
+    <!-- ... -->
+    <my-element></my-element>
+  ```
```

Comment:
not sure how to emphasize it, external DCE along with modules resolution is a requirement for mature libs. Without it there is no competition with current JS stack.

Reply:
The "external" is not limited to URL. It could be the in-page reference. To make it valuable in complex encapsulation cases, a subject for scoping. I.e. reference shall be resolved in the "scope" accounting the namespace(s) and encapsulation.

Reply:
I don’t count elsewhere on the page as “external”. With server side tooling, you _might_ not get “external” declarative custom elements in an initial draft, much like we didn’t with declarative shadow DOM. Where implementors draw the lines of shippable will not always be where we and consumers would.

Reply:
We are going away from server side tooling step by step. CDN became a host for many web components. Importmaps making the use of it. The client side caching and pre-compilation makes build time optimization obsolete. The next logical step is make externalized DCE module as a standard proposal.